.travis.yml

```diff
@@ -1,10 +1,10 @@
 language: python
 
 python:
-    - "2.7"
```

Comment:
I am not sure if cutting 2.7 will block the removal of the remote module on the Avocado VT side. There was a proposal by @sathnaga and a few other maintainers in the past to drop support of 2.7 there and move on but I am not sure about the state of this proposal.

Reply:
Unless it's breaking anything I'd also prefer to keep aexpect at least buildable on python 2.7. Is there a strict reason to abandon it?

Reply:
> Unless it's breaking anything I'd also prefer to keep aexpect at least buildable on python 2.7. Is there a strict reason to abandon it?

There's current breakage wrt Python 2.7 syntax compatibility, for instance:

```
Compiling /builddir/build/BUILDROOT/python-aexpect-1.5.1-1.20200525gitbe9be4b.fc31.x86_64/usr/lib/python2.7/site-packages/aexpect/remote_door.py ...
  File "/usr/lib/python2.7/site-packages/aexpect/remote_door.py", line 127
    def run_remote_util(session, utility, function, *args, verify=None, detach=False, **kwargs):
                                                                ^
SyntaxError: invalid syntax
```

Reply:
Yes, we had this breakage at the time of the PR and decided not to go around it but let the CI fail. I guess you have a point that this concerns building as well though. I guess worst case we can drop the remote door module through a bit of magic for a 2.7 SPEC even though the changes dropping it from the SPEC were already done and the progress there will be lost. I am still not sure why keeping Python 2.7 is still so relevant. @ldoktor Do you mind explaining some of the use cases for Python 2.7 here? I guess I have to ask the same in relation to Avocado VT where you previously insisted we should also keep such compatibility. If is is just a few users relying on this could they maybe clone the repos in their previous version to keep using them?

Reply:
The reason for me is users. Yes, they can checkout oldish versions, but it's easier to just keep it working unless it becames a big burden. Addressing the issue should be quite simple as we are receiving full kwargs.

Anyway (similarly to Avocado-vt) if the fix becomes too ugly or extensive, I won't block py2 removal.

As for the py2 usage, it's been in use for old-rhel on new-rhel testing. You are right, it can be replaced with oldish Avocado-vt in guests, but these do not contain useful fixes and maintaining multiple branches downstream (for each team as teams usually only cooperate upstream) is not as convenient as keeping py2 alive.

Reply:
I think all of this makes sense. The question that remains to answer together then is where to draw the line of too complex and expensive. I don't mean here specifically but in the long run. I guess this could pass the bar.

Reply:
> The reason for me is users. Yes, they can checkout oldish versions, but it's easier to just keep it working unless it becames a big burden. Addressing the issue should be quite simple as we are receiving full kwargs.
> 
> Anyway (similarly to Avocado-vt) if the fix becomes too ugly or extensive, I won't block py2 removal.
> 
> As for the py2 usage, it's been in use for old-rhel on new-rhel testing. You are right, it can be replaced with oldish Avocado-vt in guests, but these do not contain useful fixes and maintaining multiple branches downstream (for each team as teams usually only cooperate upstream) is not as convenient as keeping py2 alive.

Python 2 is D-E-A-D. Users on Python 2 can get older versions. I've talked to heavy users of Avocado + Avocado-VT + aexpect on the Python 2 situation (EL 7), and they have their versions pinned for compatibility.

@ldoktor if you find this important enough to send fixes and maintain Python 2 support at all times, then we can keep it.  If not, it must go.